File: Taller 03/Python/tv.py
class TV:
    num_tv = 0

    def __init__(self, marca, estado):
        TV.num_tv += 1
        self.marca = marca
        self.estado = estado
        self.canal = 1
        self.precio = 500
        self.volumen = 1
        self.control = None

    def volumen_up(self):
        if self.estado and self.volumen < 7:
            self.volumen += 1

    def canal_up(self):
        if self.estado and self.canal < 120:
            self.canal += 1

    def volumen_down(self):
        if self.estado and self.volumen > 0:
            self.volumen -= 1

    def canal_down(self):
        if self.estado and self.canal > 1:
            self.canal -= 1

    def turn_on(self):
        self.estado = True

    def turn_off(self):
        self.estado = False

    @staticmethod
    def get_num_tv():
        return TV.num_tv

    @staticmethod
    def set_num_tv(num_tv):
        TV.num_tv = num_tv

    def get_marca(self):
        return self.marca

    def set_marca(self, marca):
        self.marca = marca

    def get_canal(self):
        return self.canal

    def set_canal(self, canal):
        if 1 <= canal <= 120:
            self.canal = canal

    def get_precio(self):
        return self.precio

    def set_precio(self, precio):
        self.precio = precio

    def get_volumen(self):
        return self.volumen

    def set_volumen(self, volumen):
        if 0 <= volumen <= 7:
            self.volumen = volumen

    def get_control(self):
        return self.control

    def set_control(self, control):
        self.control = control

File: Taller 03/Java/src/main/java/taller3/televisores/TV.java
package taller3.televisores;

public class TV {
	private Marca marca;
	private int canal = 1;
	private int precio = 500;
	private boolean estado;
	private int volumen = 1;
	private Control control;
	private static int numTV = 0;
	
	public TV(Marca marca, boolean estado){
		numTV ++;
		this.marca = marca;
		this.estado = estado;
	}
	
	public void volumenUp() {
		if(estado == true) {
			if(volumen < 7) {
				volumen++;
			}
		}
	}
	
	public void canalUp() {
		if(estado == true) {
			if(canal < 120) {
				canal++;
			}
		}
	}
	
	public void volumenDown() {
		if(estado == true) {
			if(volumen > 0) {
				volumen--;
			}
		}
	}
	
	public void canalDown() {
		if(estado == true) {
			if(canal > 1) {
				canal--;
			}
		}
	}
	
	public void turnOn() {
		estado = true;
	}
	
	public void turnOff() {
		estado = false;
	}
	
	public boolean getEstado() {
		return estado;
	}
	
	public static int getNumTV() {
		return numTV;
	}
	
	public static void setNumTV(int numTV) {
		TV.numTV = numTV;
	}

	public Marca getMarca() {
		return marca;
	}
	
	public void setMarca(Marca marca) {
		this.marca = marca;
	}
	
	public int getCanal() {
		return canal;
	}
	
	public void setCanal(int canal) {
		if(canal <= 120) {
			if(canal >= 0) {
				this.canal = canal;
			}
		}
	}
	
	public int getPrecio() {
		return precio;
	}
	
	public void setPrecio(int precio) {
		this.precio = precio;
	}
	
	public int getVolumen() {
		return volumen;
	}
	
	public void setVolumen(int volumen) {
		if(volumen <= 7) {
			if(volumen >= 0) {
				this.volumen = volumen;
			}
		}
	}
	
	public Control getControl() {
		return control;
	}
	
	public void setControl(Control control) {
		this.control = control;
	}
	
	
}
